lambdas/subset_granules.py

```diff
@@ -1,26 +1,132 @@
+        r"(?P<acquisitionDay>[0-9]{2})"
+        r"_"
+        r"(?P<processingYear>[0-9]{4})"
+        r"(?P<processingMonth>[0-9]{2})"
+        r"(?P<processingDay>[0-9]{2})"
+        r"_"
+        r"(?P<collectionNumber>\w{2})"
+        r"_"
+        r"(?P<collectionCategory>\w{2})$"
+    )
+
+    for pattern in [collection_pattern, precollection_pattern]:
+        match = re.match(pattern, sceneid, re.IGNORECASE)
+        if match:
+            meta: Dict[str, Any] = match.groupdict()
+            break
+
+    meta["scene"] = sceneid
+    if meta.get("acquisitionJulianDay"):
+        date = datetime.datetime(
+            int(meta["acquisitionYear"]), 1, 1
+        ) + datetime.timedelta(int(meta["acquisitionJulianDay"]) - 1)
+
+        meta["date"] = date.strftime("%Y-%m-%d")
+    else:
+        meta["date"] = "{}-{}-{}".format(
+            meta["acquisitionYear"], meta["acquisitionMonth"], meta["acquisitionDay"]
+        )
+
+    collection = meta.get("collectionNumber", "")
+    if collection != "":
+        collection = "c{}".format(int(collection))
+
+    return meta
+
+
+def build_landsat_s3_path(granule):
+    scene_id = granule["landsat_product_id"]
+    if len(scene_id) < 40:
+        pass
+    meta = landsat_parse_scene_id(scene_id)
+    if meta["sensor"] != "C":
+        raise NameError("USGS Landsat Scene sensor is not OLI-TIRS.\n")
+    ls_s3_path = (
+        "s3://usgs-landsat/collection"
+        + meta["collectionNumber"]
+        + "/level-"
+        + meta["processingCorrectionLevel"][1:2]
+        + "/standard/oli-tirs/"
+        + meta["acquisitionYear"]
+        + "/"
+        + meta["path"]
+        + "/"
+        + meta["row"]
+        + "/"
+        + scene_id
+        + "/"
+    )
+    granule["s3_location"] = ls_s3_path
+    return
+
+
 def select_granules(start_date, end_date, bucket, key):
+    date_format = "%Y-%m-%d %H:%M:%S"
+    dt_start_date = datetime.datetime.strptime(start_date, date_format)
+    dt_end_date = datetime.datetime.strptime(end_date, date_format)
+    assert dt_start_date < dt_end_date
     s3 = boto3.client("s3")
     response = s3.select_object_content(
         Bucket=bucket,
         Key=key,
         ExpressionType="SQL",
-        Expression="select * from S3Object[*].available_products[*] s where"
-        " TO_TIMESTAMP(s.date_acquired, 'y/MM/dd') BETWEEN"
-        f" TO_TIMESTAMP('{start_date}', 'y/MM/dd') AND"
-        f" TO_TIMESTAMP('{end_date}', 'y/MM/dd') AND"
-        " s.processing_level = '1' AND"
+        Expression="select * from S3Object s where"
+        " TO_TIMESTAMP(s.date_acquired,'yyyy-MM-dd HH:mm:ss') BETWEEN"
+        f" TO_TIMESTAMP('{start_date}','yyyy-MM-dd HH:mm:ss') AND"
+        f" TO_TIMESTAMP('{end_date}','yyyy-MM-dd HH:mm:ss') AND"
+        " s.processing_level_short = '1' AND"
         " s.sensor_id = 'OLI_TIRS' AND"
-        " s.spacecraft_id = 'LANDSAT_8' AND"
-        " s.product_id LIKE '%_T1'",
+        " s.landsat_product_id LIKE '%_T1' AND"
```

Comment:
Include optional Landsat 8/9 filter to support Landsat 9 reprocessing use case.

Reply:
Added environment variable to set Landsat 8 or 9 for historical processing.